libs/designer-ui/src/lib/utils/utils.ts

```diff
@@ -354,8 +354,8 @@ export const getConnectorCategoryString = (connectorId: string): string => {
   return isBuiltInConnector(connectorId) ? builtInText : isCustomConnector(connectorId) ? customText : azureText;
 };
 
-export const convertUIElementNameToAutomationId = (uiElementName: string): string => {
-  return uiElementName?.replace(/\W/g, '_')?.toLowerCase();
+export const convertUIElementNameToAutomationId = (uiElementName: string | undefined): string => {
+  return uiElementName ? uiElementName.replace(/\W/g, '_')?.toLowerCase() : '';
```

Comment:
How is this changing the behavior, we should be returning undefined when there is no UiElementName and not empty string.

Reply:
i tested this code in a small hello world app to verify (please see screenshot below if needed) -- previously, if `uiElementName` was an empty string, we would return an empty string. In the new behavior also, if uiElementName is empty string or undefined, we return an empty string. So the behavior is unchanged. Previously, uiElementName couldn't be `undefined` because it was defined as `string` type, now it can

Test code:
![image](https://github.com/Azure/LogicAppsUX/assets/10837129/7788e50e-972c-447c-b0ec-af7cc3c5e06d)

Result:
![image](https://github.com/Azure/LogicAppsUX/assets/10837129/85958924-2bfb-4403-8a54-6b158b700264)

Reply:
We want to send back undefined when uiElement is undefined, so that no new attributes are added in the html component, if it is empty string attribute will stil be added but with empty string. This is what we wanted to avoid int eh first PR done on this.
you can defined uiElement to be string | undefined but this code still works - uiElementName?.replace(/\W/g, '_')?.toLowerCase();

Reply:
Removed this change